### PR TITLE
feat: Add endpoints for organisation page [DEV-3848]

### PR DESCRIPTION
**Pull request by lampkin-diet**

### CODE REVIEW

**Comment by ankurdotb** [comment]
Task linked: [DEV-3848 Implement organisation details endpoints in customer table and build API](https://app.clickup.com/t/6600954/DEV-3848)

File: src/controllers/admin/organisation.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+export class OrganisationController {
+	static organisationUpdatevalidator = [
+		check('name').
+			optional().
+			isString().
+			withMessage('name should be a valid string'),
+		check('email').
+			optional().
+			isString().
```
Comment (DaevMithran) [inline src/controllers/admin/organisation.ts:20]: Better to use `isEmail` here

File: src/controllers/admin/organisation.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	 *         $ref: '#/components/schemas/NotFoundError'
+	 * */
+	@validate
+	async update(request: Request, response: Response) {
+		const { name, email, description } = request.body;
+		try {
+			const customer = await CustomerService.instance.update(response.locals.customer.customerId, name, email, description);
+			const paymentAccount = await PaymentAccountService.instance.find({ customer: customer });
+
+			if (!customer || !paymentAccount) {
```
Comment (DaevMithran) [inline src/controllers/admin/organisation.ts:77]:
```suggestion
			if (!customer || paymentAccount.length === 0) {
```

File: src/controllers/admin/organisation.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	 *         $ref: '#/components/schemas/InternalError'
+	 *       404:
+	 *         $ref: '#/components/schemas/NotFoundError'
+	 */ 
+	async get(request: Request, response: Response) {
+		try {
+			const customer = response.locals.customer;
+			const paymentAccount = await PaymentAccountService.instance.find({ customer: response.locals.customer });
+
+			if (!customer || !paymentAccount) {
```
Comment (DaevMithran) [inline src/controllers/admin/organisation.ts:125]:
```suggestion
			if (!customer || paymentAccount.length === 0) {
```

**Review by DaevMithran — CHANGES_REQUESTED** [review]

**Review by ankurdotb — APPROVED** [review]

**Pull request merged**